### PR TITLE
fix for #7640 (formating related issue in javadoc)

### PR DESCRIPTION
**Pull request by naren2605**
fixes issue mentioned here #7640 :
- @return is always treated as block tag , since java 16 return can be both inline and block tag as mentioned in the the [spec](https://docs.oracle.com/en/java/javase/22/docs/specs/javadoc/doc-comment-spec.html#Where%20Tags%20Can%20Be%20Used)
- this also happens with any inline tag which is not present in list mentioned [here](https://github.com/apache/netbeans/blob/master/java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java#L4825)

fix:
- fix provided to check inline tags with inline tag prefix instead of checking with a list , as this list can grow and would impact other inline tags which are not there in the [list](https://github.com/apache/netbeans/blob/master/java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java#L4825)

### CODE REVIEW

File: java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java [hunk 1]
```diff
@@ -4788,127 +4788,135 @@ private void reformatComment() {
...
             int maxParamNameLength = 0;
             int maxExcNameLength = 0;
             int initTextEndOffset = Integer.MAX_VALUE;
+            final int ACTION_TYPE_ADD_BLANK_LINE = 0;
+            final int ACTION_TYPE_ADD_NEW_LINE = 1;
+            final int ACTION_TYPE_ALIGN_PARAMS = 2;
+            final int ACTION_TYPE_ALIGN_RETURN = 3;
+            final int ACTION_TYPE_ALIGN_EXCEPTIONS = 4;
+            final int ACTION_TYPE_NO_FORMAT = 5;
+            final int ACTION_TYPE_FORMAT = 6;
```
Comment (mbien) [inline java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java:4803]: this looks like a candidate for an `enum`

File: java/java.source.base/test/unit/src/org/netbeans/modules/java/source/save/FormatingTest.java [hunk 2]
```diff
@@ -5201,6 +5201,51 @@ public void testJavadoc() throws Exception {
                 + "}\n";
         reformat(doc, content, golden);
 
+        
+        content ="package hierbas.del.litoral;\n" +
+                    "\n" +
```
Comment (mbien) [inline java/java.source.base/test/unit/src/org/netbeans/modules/java/source/save/FormatingTest.java:5206]: feel free to bump the java version to 17 to unlock text blocks.
replace
https://github.com/apache/netbeans/blob/08339224a1417114ef2838fccbc5fcfeeb992c1e/java/java.source.base/nbproject/project.properties#L21
with `java.release=17`

**Comment by lahodaj** [comment]
I am sorry, but it seems this patch contains way too many "drive-by" changes to allow a reasonable review. I probably could accept having constants instead of hardcoded number, but changing `lastWSOffset` to `lastWhiteSpaceOffset` does not seem very useful to me. I would suggest to cleanup the patch, removing as many of drive-by changes (changes that are not necessary for the functionality of the patch).

**Comment by mbien** [comment]
@lahodaj its two commits, the first refactors, the second fixes the issue. We do allow the two commit approach, otherwise nobody would ever refactor anything ;)

But I agree that the renaming could be probably reduced a bit.

**Comment by naren2605** [comment]
> I am sorry, but it seems this patch contains way too many "drive-by" changes to allow a reasonable review. I probably could accept having constants instead of hardcoded number, but changing `lastWSOffset` to `lastWhiteSpaceOffset` does not seem very useful to me. I would suggest to cleanup the patch, removing as many of drive-by changes (changes that are not necessary for the functionality of the patch).

hi @lahodaj , @mbien  thanks for taking a look in to pr ,  i have added minor refactoring changes to improve readability of code in the same method where fix related changes exist(while i was debugging the issue), just following this [clean code principle](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html). as @mbien  mentioned i have added refactoring changes in one commit(where tests are passing before adding fix) and fix as head commit(with new tests). i would like to summarise refactoring changes as below. take a look and let me know which and all refactoring changes you see doesn't add value, will revert them accordingly.

1. introduced  variables like `ACTION_TYPE_ADD_BLANK_LINE ` and  `STATE_INITIAL_TEXT` as constants to remove code smell - *magic numbers* (as you are ok with this and @mbien was suggesting to add enums, will enumize these)
2. variable like currNWSPos, lastNWSPos , currWSPos, lastWSPos as NWS and WS here was bit cryptic i have renamed these variables accordingly to improve readability(will revert these changes as you suggested these doesn't seem to be adding value)
3. renamed variable names from `toAdd ` to `marker` , `marker` noun suited better as we are adding marked position and action  to `LinkedList<Pair<Integer, Integer>> marks = new LinkedList<Pair<Integer, Integer>>()` , renamed  variable`checkOffset` verb to `markedOffset` noun as we are  retrieving markedOffset from  `marks`  list
4. few marked variables like `offset` as final , as current method is ~700 lines long and multiple variables and states  are mutating, just marked some variables like these to `final` which are used as readonly values with the intent to improve readability.

let me know @lahodaj if you don't see value add on point's 3,4 will go ahead and revert them.

File: java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java [hunk 3]
```diff
@@ -4831,19 +4831,19 @@ private void reformatComment() {
                             nlAdd = null;
                             String tokenText = javadocTokens.token().text().toString();
                             int newState;
-                            if (JDOC_PARAM_TAG.equalsIgnoreCase(tokenText)) {
+                            if (hasInlineTagPrefix(text,tokenText,javadocTokens.offset()-offset)) {
```
Comment (mbien) [inline java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java:4841]: space after `,` and before/after `-`, analog to other usages in this file

**Review by mbien — COMMENTED** [review]
looks good to me, added some minor comments mostly about formatting issues

please sqush into two commits, one with the refactoring, the other with the fix and force push.

Once done we can run CI and get a dev-build from it for a manual smoke test.

**Review by mbien — APPROVED** [review]
looks good to me - thanks. A tested this a little bit and all inline tags are now left untouched by the formatter.

**Review by dbalek — APPROVED** [review]
Looks fine. All formating tests are passing.

**Review by lahodaj — APPROVED** [review]
Seems OK.

**Comment by naren2605** [comment]
@mbien when would this be merged?

**Comment by mbien** [comment]
three approvals, java tests ran and are green, commit header/email address looks valid, fix and refactoring is kept separate -> merging

@naren2605 thanks and congrats on your first NetBeans contribution ;)

**Pull request merged**